[SR] Add callout for managed repository and prevent deletion from UI

jen-huang opened the pull request:
Relates to https://github.com/elastic/cloud/issues/29439

## Summary
This PR adjusts the UI logic when a repository is detected to be a Cloud-managed repository. We want to:
* Surface messaging to the user that changing the Cloud-managed repository may have unwanted effects (aka - make sure they know what they are doing!)
* Prevent user from accidentally deleting this repository from the UI

## Testing instructions
1. Add `cluster.metadata.managed_repository` setting via Console:
```
PUT /_cluster/settings
{
  "persistent": {
    "cluster.metadata.managed_repository": "found-snapshots"
  }
}
```
2. Create a repository with the same name as your setting value (`found-snapshots`)

Note: Neither step is necessary when this goes live on Cloud, as both the repository and the setting will already exist, we are just mocking the environment locally here 🙂

## Copy feedback
I would love some help from @zanbel / @yaronp68 / @gchaps on refining the messaging of these two strings. You can see them in action in the screenshots below.

Short version:
```
This is a managed repository. Proceed with caution!
```
Verbose version:
```
This is a managed repository. Changing these settings may have adverse effects. Proceed with caution!
```

## Screenshots
In the table, a **Managed** badge appears next to the name, and deletion is disabled (user is not able to select the row and trash can icon is disabled):

![image](https://user-images.githubusercontent.com/1965714/58611111-e57b9c00-8262-11e9-865c-8fa4512b151e.png)

In repository details, Remove button is disabled, and a warning callout appears:

![image](https://user-images.githubusercontent.com/1965714/58611032-93d31180-8262-11e9-97db-ff3f8c6e65e5.png)

When editing the managed repository, another warning callout appears, and the Save button is treated differently:

![image](https://user-images.githubusercontent.com/1965714/58611009-7b62f700-8262-11e9-8e81-e0208cd03231.png)
![image](https://user-images.githubusercontent.com/1965714/58611021-84ec5f00-8262-11e9-9e48-b15ee7b84a68.png)

elasticmachine commented:
Pinging @elastic/es-ui

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/14852/)

gchaps commented:
Going off what @cjcenizal and @alisonelizabeth said, here are some suggestions:

**Details pane**

This is a managed repository used by other systems.  Any changes you make might affect how these systems operate.  

**Edit pane**

This is a managed repository. Changing this repository might affect other systems that use it. Proceed with caution.

**Button label**

I played around with the button label a bit.  I'm not really a fan of including parens in a button label.  Maybe using the warning at the top and this button label would be enough?

Save repository changes

**Tooltip**

You cannot delete a managed repository.

cjcenizal commented:
@gchaps You're right, the warning at the top is probably enough. I don't feel too strongly about the copy inside the button, as long as it is styled differently so that users pause before instinctively clicking it.

jen-huang commented:
I've adjusted the copy, Save button styling, and added tooltips to disabled delete actions for a managed repository. Screenshots in main description has been updated.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/16093/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/16104/)
